migration-engine/migration-engine-tests/tests/migrations/mysql.rs

```diff
@@ -308,3 +308,54 @@ fn default_current_timestamp_precision_follows_column_precision(api: TestApi) {
         .send_sync()
         .assert_migration("01init", |migration| migration.assert_contents(expected_migration));
 }
+
+#[test_connector(tags(Mysql))]
+fn mysql_apply_migrations_errors_gives_the_failed_sql(api: TestApi) {
+    let dm = "";
+    let migrations_directory = api.create_migrations_directory();
+
+    let migration = r#"
+        CREATE TABLE `Cat` ( id INTEGER PRIMARY KEY );
+
+        DROP TABLE `Emu`;
+
+        CREATE TABLE `Emu` (
+            size INTEGER
+        );
+    "#;
+
+    let migration_name = api
+        .create_migration("01init", dm, &migrations_directory)
+        .draft(true)
+        .send_sync()
+        .modify_migration(|contents| {
+            contents.clear();
+            contents.push_str(migration);
+        })
+        .into_output()
+        .generated_migration_name
+        .unwrap();
+
+    let err = api
+        .apply_migrations(&migrations_directory)
+        .send_unwrap_err()
+        .to_string()
+        .replace(&migration_name, "<migration-name>");
+
+    let expectation = expect![[r#"
+        A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
+
+        Migration name: <migration-name>
+
+        Database error code: 1051
+
+        Database error:
+        Unknown table 'mysql_apply_migrations_errors_gives_the_failed_sql.Emu'
+
+        Please check the query number 1 from the migration file.
```

Comment:
Purely from the phrasing and the SQL migration, I would expect the first CREATE TABLE to be query number 1, but the error message is zero-indexed. Do you think it would be clearer the other way? Or maybe we can find a phrasing that makes it clear if this is zero-indexed or one-indexed.